packages/spacecat-shared-utils/test/log-wrapper.test.js

```diff
@@ -0,0 +1,127 @@
+            body: message,
+          }],
+        },
+      },
+      log: {
+        info: sinon.spy(),
+        error: sinon.spy(),
+        debug: sinon.spy(),
+        warn: sinon.spy(),
+      },
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should wrap log methods to include jobId when jobId is present', async () => {
+    const wrappedFn = logWrapper(mockFnFromSqs);
+
+    await wrappedFn(message, mockContext);
+
+    mockContext.contextualLog.info('Test log');
+    const logArgs = mockContext.log.info.getCall(0).args[0];
+    expect(containsKeyValue(logArgs, 'jobId', message.jobId)).to.be.true;
+  });
+
+  it('should not wrap log methods when jobId is missing', async () => {
+    const wrappedFn = logWrapper(mockFnFromSqs);
+
+    await wrappedFn({ message: 'This contains no jobId' }, mockContext);
+
+    mockContext.contextualLog.info('Test log');
+    const logArgs = mockContext.log.info.getCall(0).args[0];
+    expect(containsKeyValue(logArgs, 'jobId', undefined)).to.be.false;
+    expect(logArgs).to.equal('Test log');
+  });
+
+  it('should call the original function with the provided message and context, and update the context with contextualLog object', async () => {
+    const wrappedFn = logWrapper(mockFnFromSqs);
+
+    await wrappedFn(message, mockContext);
+
+    // Verify the original function is called with the correct parameters
+    expect(mockFnFromSqs.calledWith(message, mockContext)).to.be.true;
+
+    // Verify the context is updated correctly
+    expect(mockContext).to.have.property('contextualLog');
+    expect(mockContext.contextualLog).to.be.an('object');
+  });
+
+  it('should call log methods with correct parameters when jobId is present', async () => {
+    const wrappedFn = logWrapper(mockFnFromSqs);
+
+    await wrappedFn(message, mockContext);
+
+    mockContext.contextualLog.info('Info log');
+    mockContext.contextualLog.error('Error log');
+    mockContext.contextualLog.debug('Debug log');
+    mockContext.contextualLog.warn('Warn log');
+
+    expect(containsKeyValue(mockContext.log.info.getCall(0).args[0], 'jobId', message.jobId)).to.be.true;
+    expect(containsKeyValue(mockContext.log.error.getCall(0).args[0], 'jobId', message.jobId)).to.be.true;
+    expect(containsKeyValue(mockContext.log.debug.getCall(0).args[0], 'jobId', message.jobId)).to.be.true;
+    expect(containsKeyValue(mockContext.log.warn.getCall(0).args[0], 'jobId', message.jobId)).to.be.true;
+  });
+
+  it('should call log methods with correct parameters when jobId is missing', async () => {
+    const wrappedFn = logWrapper(mockFnFromSqs);
+
+    await wrappedFn({}, mockContext);
+
+    mockContext.contextualLog.info('Info log');
+    mockContext.contextualLog.error('Error log');
+    mockContext.contextualLog.debug('Debug log');
+    mockContext.contextualLog.warn('Warn log');
+
+    expect(containsKeyValue(mockContext.log.info.getCall(0).args[0], 'jobId', undefined)).to.be.false;
```

Comment:
Are these `containsKeyValue` checks false because the object doesn't have `jobId`, or because its value does not equal `undefined`? I think a more specific check might be needed here.